src/parser/mod.rs

```diff
@@ -1269,6 +1269,50 @@ impl<'a> Parser<'a> {
             _ => self.expected("an expression", next_token),
         }?;
 
+        if dialect_of!(self is MsSqlDialect) {
+            // Convert `CompositeAccess` to `CompositeFunction` (MSSQL doesn't support `CompositeAccess` syntax)
+            // ```sql
+            // SELECT (SELECT ',' + name FROM sys.objects  FOR XML PATH(''), TYPE).value('.','NVARCHAR(MAX)')
```

Comment:
is it possible to represent this syntax as the existing `Expr::JsonAccess`? I'm assuming that isn't the case since otherwise the the feature will be already be supported. If not its not super clear to me what exactly is the composite access we're introducing with this PR - My assumption is that its adding support for syntax like `<arbitrary-expr>.<arbitrary-expr>.<arbitrary-expr>...` and so on where `arbitrary-expr` can be any expr including a function call. Like in the example `SELECT CONVERT(XML,'<Book>abc</Book>').value('.','NVARCHAR(MAX)')`

Reply:
`CompositeFunction` is similar to `CompositeAccess` but supports method call syntax, that supports method calls on subquery results (must be `XML` data type in MSSQL).
Maybe it would be better to make  `CompositeFunction` a generic feature?